src/Stratis.Bitcoin.Tests/Consensus/ConsensusManagerTest.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Moq;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Configuration.Logging;
+using Stratis.Bitcoin.Configuration.Settings;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Tests.Common;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.Consensus
+{
+    public class ConsensusManagerTest
+    {
+        public class TestContext
+        {
+            public Network Network = Network.RegTest;
+            public Mock<IBlockValidator> ChainedHeaderValidatorMock = new Mock<IBlockValidator>();
+            public Mock<ICheckpoints> CheckpointsMock = new Mock<ICheckpoints>();
+            public Mock<IBlockPuller> BlockPullerMock = new Mock<IBlockPuller>();
+
+            public Mock<IChainState> ChainStateMock = new Mock<IChainState>();
+            public Mock<IConsensusRules> ConsensusRulesMock = new Mock<IConsensusRules>();
+            public Mock<IConnectionManager> ConnectionManagerMock = new Mock<IConnectionManager>();
+
+            public Mock<IFinalizedBlockHeight> FinalizedBlockMock = new Mock<IFinalizedBlockHeight>();
+            public ConsensusSettings ConsensusSettings = new ConsensusSettings(new NodeSettings(Network.RegTest));
+
+            internal ChainedHeaderTree ChainedHeaderTree;
+
+            public ConsensusManager ConsensusManager;
+
+            internal ConsensusManager CreateConsensusManager()
+            {
+                this.ConsensusManager = new ConsensusManager(this.Network, new ExtendedLoggerFactory(), this.ChainStateMock.Object, 
+                    this.ChainedHeaderValidatorMock.Object, this.CheckpointsMock.Object, this.ConsensusSettings, this.BlockPullerMock.Object,
+                    this.ConsensusRulesMock.Object, this.FinalizedBlockMock.Object, this.ConnectionManagerMock.Object );
+
+                this.ChainedHeaderTree = this.ConsensusManager.GetMemberValue("chainedHeaderTree") as ChainedHeaderTree;
+
+                return this.ConsensusManager;
+            }
+
+            public ChainedHeader ExtendAChain(int count, ChainedHeader chainedHeader = null)
```

Comment:
wtf? This code already exist elsewhere but it is better, why would you create a copy of that is just worse? Why not use the code from CHT tests?

Reply:
I think its not pushed yet, but yeah probably we need some shared test context, I will let the testers do that.